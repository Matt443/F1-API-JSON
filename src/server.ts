import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}