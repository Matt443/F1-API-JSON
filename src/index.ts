import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
