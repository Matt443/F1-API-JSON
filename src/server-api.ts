import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
