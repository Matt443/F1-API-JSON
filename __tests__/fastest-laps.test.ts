import { getFastestLaps } from "../src/server";

test("fastest laps", async () => {
    expect(await getFastestLaps()).toMatchSnapshot();
});
