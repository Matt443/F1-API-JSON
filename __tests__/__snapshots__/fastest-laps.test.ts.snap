// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fastest laps 1`] = `
[
  {
    "car": "Red Bull Racing Honda RBPT",
    "driver": "Max Verstappen",
    "grandPrix": "Bahrain",
    "time": "1:32.608",
  },
  {
    "car": "Ferrari",
    "driver": "Charles Leclerc",
    "grandPrix": "Saudi Arabia",
    "time": "1:31.632",
  },
  {
    "car": "Ferrari",
    "driver": "Charles Leclerc",
    "grandPrix": "Australia",
    "time": "1:19.813",
  },
  {
    "car": "Red Bull Racing Honda RBPT",
    "driver": "Max Verstappen",
    "grandPrix": "Japan",
    "time": "1:33.706",
  },
  {
    "car": "Aston Martin Aramco Mercedes",
    "driver": "Fernando Alonso",
    "grandPrix": "China",
    "time": "1:37.810",
  },
  {
    "car": "McLaren Mercedes",
    "driver": "Oscar Piastri",
    "grandPrix": "Miami",
    "time": "1:30.634",
  },
  {
    "car": "Mercedes",
    "driver": "George Russell",
    "grandPrix": "Emilia-Romagna",
    "time": "1:18.589",
  },
  {
    "car": "Mercedes",
    "driver": "Lewis Hamilton",
    "grandPrix": "Monaco",
    "time": "1:14.165",
  },
  {
    "car": "Mercedes",
    "driver": "Lewis Hamilton",
    "grandPrix": "Canada",
    "time": "1:14.856",
  },
  {
    "car": "McLaren Mercedes",
    "driver": "Lando Norris",
    "grandPrix": "Spain",
    "time": "1:17.115",
  },
  {
    "car": "Aston Martin Aramco Mercedes",
    "driver": "Fernando Alonso",
    "grandPrix": "Austria",
    "time": "1:07.694",
  },
  {
    "car": "Ferrari",
    "driver": "Carlos Sainz",
    "grandPrix": "Great Britain",
    "time": "1:28.293",
  },
  {
    "car": "Mercedes",
    "driver": "George Russell",
    "grandPrix": "Hungary",
    "time": "1:20.305",
  },
  {
    "car": "Red Bull Racing Honda RBPT",
    "driver": "Sergio Perez",
    "grandPrix": "Belgium",
    "time": "1:44.701",
  },
  {
    "car": "McLaren Mercedes",
    "driver": "Lando Norris",
    "grandPrix": "Netherlands",
    "time": "1:13.817",
  },
  {
    "car": "McLaren Mercedes",
    "driver": "Lando Norris",
    "grandPrix": "Italy",
    "time": "1:21.432",
  },
  {
    "car": "McLaren Mercedes",
    "driver": "Lando Norris",
    "grandPrix": "Azerbaijan",
    "time": "1:45.255",
  },
  {
    "car": "RB Honda RBPT",
    "driver": "Daniel Ricciardo",
    "grandPrix": "Singapore",
    "time": "1:34.486",
  },
  {
    "car": "Alpine Renault",
    "driver": "Esteban Ocon",
    "grandPrix": "United States",
    "time": "1:37.330",
  },
  {
    "car": "Ferrari",
    "driver": "Charles Leclerc",
    "grandPrix": "Mexico",
    "time": "1:18.336",
  },
  {
    "car": "Red Bull Racing Honda RBPT",
    "driver": "Max Verstappen",
    "grandPrix": "Brazil",
    "time": "1:20.472",
  },
  {
    "car": "McLaren Mercedes",
    "driver": "Lando Norris",
    "grandPrix": "Las Vegas",
    "time": "1:34.876",
  },
  {
    "car": "McLaren Mercedes",
    "driver": "Lando Norris",
    "grandPrix": "Qatar",
    "time": "1:22.384",
  },
  {
    "car": "Haas Ferrari",
    "driver": "Kevin Magnussen",
    "grandPrix": "Abu Dhabi",
    "time": "1:25.637",
  },
]
`;
